### PR TITLE
Rkt pull

### PR DESCRIPTION
**Pull request by yifan-gu**
@vmarmol @dchen1107 @jonboulle 

Still depends #7465

### CODE REVIEW

File: pkg/kubelet/rkt/pull.go [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+	"github.com/docker/docker/pkg/parsers"
+	docker "github.com/fsouza/go-dockerclient"
+	"github.com/golang/glog"
+)
+
+const (
+	authDir = "auth.d"
+)
+
+func (r *Runtime) writeDockerAuthConfig(image string, creds docker.AuthConfiguration) error {
```
Comment (vmarmol) [inline pkg/kubelet/rkt/pull.go]: Can we add a TODO to merge this with the dockertools logic?

File: pkg/kubelet/rkt/pull.go [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+			return err
+		}
+	}
+	f, err := os.Create(path.Join(localConfigDir, authDir, registry+".json"))
+	if err != nil {
+		glog.Errorf("rkt: Cannot create docker auth config file: %v", err)
+		return err
+	}
+	defer f.Close()
+	config := fmt.Sprintf(dockerAuthTemplate, registry, creds.Username, creds.Password)
```
Comment (vmarmol) [inline pkg/kubelet/rkt/pull.go]: Will `dockerAuthTemplate` be used anywhere else? If not, can we move the definition here?

**Comment by vmarmol** [comment]
LGTM, only minor nits

File: pkg/kubelet/rkt/pull.go (hunk 2 above)
Comment (yifan-gu) [inline pkg/kubelet/rkt/pull.go]: done

File: pkg/kubelet/rkt/pull.go (hunk 1 above)
Comment (yifan-gu) [inline pkg/kubelet/rkt/pull.go]: I think it's actually belongs to rkt. I added some doc.

**Comment by vmarmol** [comment]
LGTM

**Pull request merged**